### PR TITLE
Checkout the order by changing the state to 'Paid'

### PR DESCRIPTION
**Pull request by AlirezaEm**

### CODE REVIEW

File: Routes/orders.py [hunk 1]
```diff
@@ -48,3 +48,42 @@ def delete_shopping_cart(uuid: str):
...
     return None    
+
+# POST /v1/orders/uuid/checkout : Checkout an entire shopping cart that changes the state to PAID and freezes it to go through shipment
+@router.post("/v1/orders/{uuid}/checkout")
+def checkout_shopping_cart(uuid: str, auth_token: Annotated[str, Header()]):
+    try:
+        payload = jwt.decode(auth_token, "secret", algorithms="HS256")
+        user_id: str = payload.get("user_id")
+        if user_id is None:
+            raise HTTPException(status_code=400, detail="The provided token does not have user id, please try logging in again.")
```
Comment (WillBkWhiteFoxCloud) [inline Routes/orders.py:59]: This is also a 401 Invalid token error.

File: Routes/orders.py [hunk 2]
```diff
@@ -48,3 +48,42 @@ def delete_shopping_cart(uuid: str):
...
+    except JWTError:
+        raise HTTPException(status_code=401, detail="Could not validate credentials")
+    
+    if uuid != user_id:
```
Comment (WillBkWhiteFoxCloud) [inline Routes/orders.py:63]: The "uuid" path parameter indicates the id of an order. A user can have multiple orders, so comparing order id and user id would be irrelevant.

Comment (AlirezaEm) [reply Routes/orders.py:63]: So an user can have multiple orders/shopping-carts. I wanted to check if the uuid is for the user's order, so I think for now I have to leave for later, do you agree?

Comment (AlirezaEm) [reply Routes/orders.py:63]: Or we can add 'owner_id' to our ShoppingCart model.

Comment (WillBkWhiteFoxCloud) [reply Routes/orders.py:63]: That's a great idea. User and Shopping cart have one to many relationship. Using a foreign key like owner_id is great way to ensure this relationship.

**Pull request merged**